test/event.test.js

```diff
@@ -1,22 +1,31 @@
-/* global describe, it */
+/* global describe, before, it */
 'use strict'
 
 const bindings = require('../')
-const aob = bindings
 const expect = require('chai').expect
 
+const env = process.env
+const maxIsReadyToSampleWait = 60000
+
 const evUnsampled = '2B5BD5777CA0077C734B537B64C6B969213358B9F21AA0B1B42979F5C400'
 const evSampled = '2B4FC9017BA3404828F253638A697DC7CF1A6BB1A4A544D5B98159B55501'
 
 describe('addon.event', function () {
-  const serviceKey = `${process.env.AO_TOKEN_PROD}:node-bindings-test`
+  before(function () {
+    const serviceKey = process.env.APPOPTICS_SERVICE_KEY || `${env.AO_TOKEN_STG}:node-bindings-test`
+    const endpoint = process.env.APPOPTICS_COLLECTOR || 'collector-stg.appoptics.com'
+
+    this.timeout(maxIsReadyToSampleWait)
+    const status = bindings.oboeInit({ serviceKey, endpoint })
+    // oboeInit can return -1 for already initialized or 0 if succeeded.
+    // depending on whether this is run as part of a suite or standalone
+    // either result is valid.
+    if (status !== -1 && status !== 0) {
+      throw new Error('oboeInit() failed')
+    }
 
-  it('should initialize oboe with only a service key', function () {
```

Comment:
is initializing with only a service key now tested somewhere else?

Reply:
Yes. We do it once in `test/bindings.test.js` and it always runs against host collector.appoptics.com (production). see (ba6d6b2c5d574611eab84b6602a1470985423072).

On a side note - the concept of "magic" host initialization, is something I would personally be reluctant to do, but, since it is part of `liboboe` (https://github.com/librato/oboe/blob/master/liboboe/reporter/ssl.cc) it bubbled up to the node addon and is hence tested...